server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java

```diff
@@ -0,0 +1,286 @@
+        private Connection node2Connection;
+        private Channel node2Channel;
+
+        @BeforeEach
+        void setup(DockerRabbitMQCluster cluster) throws IOException, TimeoutException {
+            node1ConnectionFactory = cluster.getRabbitMQ1().connectionFactory();
+            resilientConnection = node1ConnectionFactory.newConnection(cluster.getAddresses());
+            resilientChannel = resilientConnection.createChannel();
+            ConnectionFactory node2ConnectionFactory = cluster.getRabbitMQ2().connectionFactory();
+            node2Connection = node2ConnectionFactory.newConnection();
+            node2Channel = node2Connection.createChannel();
+        }
+
+        @AfterEach
+        void tearDown() {
+            closeQuietly(resilientConnection, resilientChannel);
+        }
+
+        @Disabled("Not sure Rabbit driver can recover at all in this case, it keeps failing even with retry")
+        @RepeatedTest(10)
+        void nodeKillingWhenProducing(DockerRabbitMQCluster cluster) throws Exception {
+            resilientChannel.exchangeDeclare(EXCHANGE_NAME, DIRECT, DURABLE);
+            resilientChannel.queueDeclare(QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, ImmutableMap.of()).getQueue();
+            resilientChannel.queueBind(QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+            int nbMessages = 20;
+            int firstBatchSize = nbMessages / 2;
+            IntStream.range(0, firstBatchSize)
+                .mapToObj(i -> asBytes(String.valueOf(i)))
+                .forEach(Throwing.consumer(
+                    bytes -> resilientChannel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES, bytes)));
+
+            cluster.getRabbitMQ1().stop();
+
+            IntStream.range(firstBatchSize, nbMessages)
+                .mapToObj(i -> asBytes(String.valueOf(i)))
+                .forEach(Throwing.consumer(
+                    bytes -> Awaitility.waitAtMost(Duration.TEN_SECONDS).until(() -> tryPublish(bytes))));
+
+
+            InMemoryConsumer consumer = new InMemoryConsumer(node2Channel);
+            node2Channel.basicConsume(QUEUE, consumer);
+
+            awaitAtMostOneMinute.until(() -> consumer.getConsumedMessages().size() == nbMessages);
+
+            List<Integer> expectedResult = IntStream.range(0, nbMessages).boxed().collect(Collectors.toList());
+            assertThat(consumer.getConsumedMessages()).containsOnlyElementsOf(expectedResult);
+        }
+
+        private boolean tryPublish(byte[] bytes) {
+            try {
+                resilientChannel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES, bytes);
+                return true;
+            } catch (Exception e) {
+                try {
+                    System.out.println("trying recover");
+                    resilientChannel.basicRecover();
+                    System.out.println("recovered");
+                } catch (Exception e1) {
+                    System.out.println("failed recover");
```

Comment:
Looks like debug statements... Could we use a real logger, if applicable, please?

Reply:
TBH, it's pending work because I didn't managed to make auto reconnection working yesterday evening, so I didn't bother using a logger here